{"metadata":{"title":"Failure","symbolKind":"associatedtype","modules":[{"name":"Amplify"}],"required":true,"role":"symbol","externalID":"s:7Amplify18InternalTaskResultP7FailureQa","fragments":[{"kind":"keyword","text":"associatedtype"},{"kind":"text","text":" "},{"kind":"identifier","text":"Failure"},{"kind":"text","text":" : "},{"kind":"typeIdentifier","preciseIdentifier":"s:7Amplify0A5ErrorP","text":"AmplifyError"}],"roleHeading":"Associated Type"},"hierarchy":{"paths":[["doc:\/\/amplify-swift.Amplify\/documentation\/Amplify","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/InternalTaskResult"]]},"kind":"symbol","schemaVersion":{"major":0,"minor":3,"patch":0},"sections":[],"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/InternalTaskResult\/Failure"},"primaryContentSections":[{"declarations":[{"tokens":[{"text":"associatedtype","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"Failure"},{"kind":"text","text":" : "},{"preciseIdentifier":"s:7Amplify0A5ErrorP","kind":"typeIdentifier","text":"AmplifyError","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AmplifyError"}],"platforms":["iOS"],"languages":["swift"]}],"kind":"declarations"}],"variants":[{"paths":["\/documentation\/amplify\/internaltaskresult\/failure"],"traits":[{"interfaceLanguage":"swift"}]}],"references":{"doc://amplify-swift.Amplify/documentation/Amplify/InternalTaskResult/Failure":{"abstract":[],"role":"symbol","required":true,"url":"\/documentation\/amplify\/internaltaskresult\/failure","fragments":[{"text":"associatedtype","kind":"keyword"},{"text":" ","kind":"text"},{"text":"Failure","kind":"identifier"},{"kind":"text","text":" : "},{"preciseIdentifier":"s:7Amplify0A5ErrorP","text":"AmplifyError","kind":"typeIdentifier"}],"title":"Failure","type":"topic","kind":"symbol","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/InternalTaskResult\/Failure"},"doc://amplify-swift.Amplify/documentation/Amplify/AmplifyError":{"navigatorTitle":[{"kind":"identifier","text":"AmplifyError"}],"role":"symbol","title":"AmplifyError","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AmplifyError","abstract":[{"type":"text","text":"Amplifyâ€™s philosophy is to expose friendly error messages to the customer that assist with debugging."},{"type":"text","text":" "},{"type":"text","text":"Therefore, failable APIs are declared to return error results with Amplify errors, which require"},{"type":"text","text":" "},{"type":"text","text":"recovery suggestions and error messages."}],"kind":"symbol","type":"topic","fragments":[{"text":"protocol","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"AmplifyError"}],"url":"\/documentation\/amplify\/amplifyerror"},"doc://amplify-swift.Amplify/documentation/Amplify/InternalTaskResult":{"navigatorTitle":[{"text":"InternalTaskResult","kind":"identifier"}],"title":"InternalTaskResult","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/InternalTaskResult","url":"\/documentation\/amplify\/internaltaskresult","fragments":[{"text":"protocol","kind":"keyword"},{"kind":"text","text":" "},{"text":"InternalTaskResult","kind":"identifier"}],"kind":"symbol","abstract":[],"type":"topic","role":"symbol"},"doc://amplify-swift.Amplify/documentation/Amplify":{"url":"\/documentation\/amplify","type":"topic","title":"Amplify","role":"collection","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify","kind":"symbol","abstract":[]}}}